addons/website_sale_digital/controllers/main.py

```diff
@@ -57,11 +57,7 @@ def portal_order_page(self, order_id=None, **post):
     ], type='http', auth='public')
     def download_attachment(self, attachment_id):
         # Check if this is a valid attachment id
-        attachment = request.env['ir.attachment'].sudo().search_read(
-            [('id', '=', int(attachment_id))],
-            ["name", "datas", "mimetype", "res_model", "res_id", "type", "url"]
-        )
-
+        attachment = request.env['ir.attachment'].sudo().search([('id', '=', int(attachment_id))])
         if attachment:
             attachment = attachment[0]
         else:
```

Comment:
```suggestion
        attachment = request.env['ir.attachment'].sudo().browse(int(attachment_id)).exists()
        if not attachment:
```
No need for a search here, a browse + exists to verify the record existence is more adequate ;).